core/src/test/java/com/github/mizool/core/concurrent/ThrowingStreamHarness.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2018-2020 incub8 Software Labs GmbH
+ * Copyright 2018-2020 protel Hotelsoftware GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mizool.core.concurrent;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import java.util.ArrayList;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import org.assertj.core.api.ThrowableAssert;
+import org.testng.annotations.DataProvider;
+
+@Slf4j
+@Builder
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ThrowingStreamHarness
+{
+    /**
+     * A task to be run/called by a future.
+     */
+    @RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+    public static final class Task implements Runnable, Supplier<Object>
+    {
+        private final int number;
+        private final Class<? extends Throwable> throwableClass;
+
+        @Override
+        public Object get()
+        {
+            run();
+            return this;
+        }
+
+        @SneakyThrows
+        @Override
+        public void run()
+        {
+            String me = "Runnable #" + number;
+
+            log.info("{} running.", me);
+            if (shouldFail())
+            {
+                log.info("{} throws Exception.", me);
```

Comment:
Perhaps change this to `log.info("{} throws {}.", me, throwableClass.getName());`?

Reply:
Sure, why not. I went with `getSimpleName()`, though.